nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/proxy/ConanProxyHelper.groovy

```diff
@@ -29,34 +27,67 @@ import static org.sonatype.nexus.common.hash.HashAlgorithm.MD5
 import static org.sonatype.nexus.common.hash.HashAlgorithm.SHA1
 import static org.sonatype.nexus.common.hash.HashAlgorithm.SHA256
 import static org.sonatype.nexus.common.hash.HashAlgorithm.SHA512
-import static org.sonatype.nexus.repository.storage.MetadataNodeEntityAdapter.P_NAME
-import static org.sonatype.repository.conan.internal.common.v1.ConanRoutes.getCoords
+import static org.sonatype.repository.conan.internal.metadata.ConanMetadata.DIGEST
+import static org.sonatype.repository.conan.internal.metadata.ConanMetadata.GROUP
+import static org.sonatype.repository.conan.internal.metadata.ConanMetadata.PATH
+import static org.sonatype.repository.conan.internal.metadata.ConanMetadata.PROJECT
+import static org.sonatype.repository.conan.internal.metadata.ConanMetadata.STATE
+import static org.sonatype.repository.conan.internal.metadata.ConanMetadata.VERSION
 
 /**
  * @since 0.0.1
  */
 class ConanProxyHelper
 {
+  public static final List<AssetKind> DOWNLOAD_ASSET_KINDS = ImmutableList.of(AssetKind.DOWNLOAD_URL, AssetKind.DIGEST, AssetKind.CONAN_PACKAGE_SNAPSHOT)
+
   public static final List<HashAlgorithm> HASH_ALGORITHMS = ImmutableList.of(SHA256, SHA1, SHA512, MD5)
 
-  static String buildAssetPath(final Context context) {
-    AssetKind assetKind = context.getAttributes().require(AssetKind.class)
-    ConanCoords conanCoords = getCoords(context)
-    return buildAssetPathFromCoords(conanCoords, assetKind)
-  }
+  public static final Map<String, AssetKind> ASSET_KIND_FILENAMES =
+      Collections.unmodifiableMap(AssetKind.values().collectEntries { [it.filename, it] })
 
-  static String buildAssetPathFromCoords(ConanCoords conanCoords, AssetKind assetKind) {
-    if (AssetKind.DOWNLOAD_URL == assetKind ||
-        AssetKind.CONAN_PACKAGE_SNAPSHOT == assetKind ||
-        AssetKind.DIGEST == assetKind
-    ) {
-      return ConanCoords.getRecipePathWithPackages(conanCoords, assetKind.getFilename())
+  static String getProxyAssetPath(final ConanCoords conanCoords, final AssetKind assetKind) {
+    if (DOWNLOAD_ASSET_KINDS.contains(assetKind)) {
+      return getRecipePathWithPackages(conanCoords, assetKind)
     }
-    return ConanCoords.getPath(conanCoords) + "/" + assetKind.getFilename()
+    return getPath(conanCoords) + "/" + assetKind.getFilename()
+  }
+
+  static ConanCoords convertFromState(final TokenMatcher.State state) {
+    return new ConanCoords(
+        state.getTokens().get(PATH),
+        state.getTokens().get(GROUP),
+        state.getTokens().get(PROJECT),
+        state.getTokens().get(VERSION),
+        state.getTokens().get(STATE),
+        state.getTokens().getOrDefault(DIGEST, null)
+    );
+  }
+
+  private static String getPath(final ConanCoords coord) {
+    return String.format("%s/%s/%s/%s/%s%s",
+        coord.getPath(),
+        coord.getGroup(),
+        coord.getProject(),
+        coord.getVersion(),
+        coord.getChannel(),
+        coord.getSha() == null ? "" : "/packages/" + coord.getSha())
```

Comment:
StringUtils.EMPTY ?

Reply:
It was moved, maybe yes, but I do not want to change behavior

Reply:
fixed https://github.com/sonatype-nexus-community/nexus-repository-conan/pull/97/commits/35de5335f2bc22babed94ccec4989d164aa63098